pkg/network/placement_configuration.go

```diff
@@ -0,0 +1,67 @@
+package network
+
+import (
+	"reflect"
+
+	cnao "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/shared"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var defaultInfraNodeAffinity = corev1.NodeAffinity{
+	RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+		NodeSelectorTerms: []corev1.NodeSelectorTerm{
+			getNodeSelectorTermForLabelExists("node-role.kubernetes.io/master"),
+		},
+	},
+}
+
+var defaultWorkloadNodeAffinity = corev1.NodeAffinity{
+	RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+		NodeSelectorTerms: []corev1.NodeSelectorTerm{
+			getNodeSelectorTermForLabelExists("node-role.kubernetes.io/worker"),
+		},
+	},
+}
+
```

Comment:
What about the other defaults?

Reply:
I went through all the components and collected these nodeSelectors and tolerations:

```
infra:

nodeSelector:
    beta.kubernetes.io/arch: amd64  # linux-bridge, bridge marker, nmstate-webhook, nmstate-handler
    node-role.kubernetes.io/master: "" # nmstate-webhook    
tolerations:
    - key: node-role.kubernetes.io/master
      operator: Exists
      effect: NoSchedule

workloads:

nodeSelector:
    beta.kubernetes.io/arch: amd64  # linux-bridge, bridge marker, nmstate-webhook, nmstate-handler, ovs-cni-amd64    
tolerations:
    - key: node-role.kubernetes.io/master # linux-bridge,  bridge marker, nmstate-handler, ovs-cni-amd64
      operator: Exists
      effect: NoSchedule
    - operator: Exists  # multus
      effect: NoSchedule
```

So we can probably use these instead of the affinities.